content/blog/2017/2017-01-18-refactoring-conditional-to-pipeline.adoc

```diff
@@ -0,0 +1,475 @@
+}
+----
+
+
+==== GIT_BRANCH
+
+*Expands to the name of the branch that was built.*
+
+*Parameters* (descriptions omitted): _all_, _fullName_.
+
+This information may or may not be exposed in Pipeline.  If you're using
+multibranch pipeline plugin `env.BRANCH_NAME` give similar basic information, but doesn't
+offer the parameters.  There are also
+link:https://issues.jenkins-ci.org/browse/JENKINS-35230[several]
+link:https://issues.jenkins-ci.org/browse/JENKINS-24141[issues]
+filed around `GIT_*` tokens in Pipeline.
+Until they are addressed fully, we can follow the pattern shown in
+link:https://github.com/jenkinsci/pipeline-examples/blob/master/pipeline-examples/gitcommit/gitcommit.groovy[pipeline-examples],
+executing a shell to get the information we need.
+
+
+.Pipeline
+[source, groovy]
+----
+GIT_BRANCH = sh(returnStdout: true, script: 'git rev-parse --abbrev-ref HEAD').trim()
+----
+
+==== CHANGES_SINCE_LAST_SUCCESS
+
+*Displays the changes since the last successful build.*
+
+*Parameters* (descriptions omitted):
+_reverse_, _format_, _changesFormat_, _showPaths_, _pathFormat_,
+_showDependencies_, _dateFormat_, _regex_, _replace_, _default_.
+
+Not only is the information provided by this token not exposed in Pipeline,
+the token has ten optional parameters, including format strings and regular expression
+searches. There are some examples,
+link:http://stackoverflow.com/questions/38084806/how-to-get-the-changes-since-the-last-successful-build-in-jenkins-pipeline[this answer on stackoverflow.com],
+that show how to get similar information in Pipeline,
+but nothing comes close to functional parity.
+Translating this token is beyond the scope of this post.
+
+== Slightly More Complex Example
+
+Let's do one more example that shows some of these conditions and tokens.
+This time we'll perform different build steps depending on what branch we're building.
+
+=== Freestyle
+
+Here's the configuration for Freestyle version
+(It's pretty long, click <<longer-pipeline, here>> to skip down to the Pipeline version):
+
+image::/images/post-images/2017-01-17/freestyle-conditional-long-full.png[Longer Freestyle Job, role="center", width=800]
+
+
+[[longer-pipeline]]
+=== Pipeline
+
+Here's the equivalent Pipeline:
+
+.Pipeline
+[pipeline]
+----
+// Script //
+properties ([
+    parameters ([
+        string (
+            defaultValue: '*',
+            description: '',
+            name : 'BRANCH_PATTERN'),
+        booleanParam (
+            defaultValue: false,
+            description: '',
+            name : 'FORCE_FULL_BUILD')
+    ])
+])
+
+node {
+    stage ('Build') {
+        checkout([$class: 'GitSCM',
+            branches: [[name: "origin/${BRANCH_PATTERN}"]],
+            doGenerateSubmoduleConfigurations: false,
+            extensions: [[$class: 'LocalBranch']],
+            submoduleCfg: [],
+            userRemoteConfigs: [[credentialsId: '9f926524-9fac-44df-99dd-53069380f286', url: 'https://github.com/bitwiseman/hermann']]])
```

Comment:
If you change `credentialsId` to something shorter (e.g. `github`), then this will prevent this whole code block from scrolling horizontally.